shared/rSharp.cpp

```diff
@@ -9,7 +9,7 @@
 
 using string_t = std::basic_string<char_t>;
 
-const wchar_t* dotnetlib_path = nullptr;
+const char_t* dotnetlib_path = nullptr;
```

Comment:
@Ian-Du I was all wrong about this when we chatted yesterday. There's a difference in character width between platforms. That's driven by nethost libraries. So, common code needs to use this char_t.

When writing platform specific code within #ifdef WINDOWS I switched to using the explicit type.